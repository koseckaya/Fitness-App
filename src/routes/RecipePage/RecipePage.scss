@import "src/styles/vars";

.scl-recipe-page {
  box-sizing: border-box;
}
