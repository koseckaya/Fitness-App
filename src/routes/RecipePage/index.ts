export { default as RecipePage } from './RecipePage';
