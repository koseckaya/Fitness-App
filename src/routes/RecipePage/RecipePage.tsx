
import { FC, useContext } from "react";
import { useParams, Link } from 'react-router-dom';
import { recipeData } from "../../data";
import { Container } from "../../components/Container";
import { Button } from "../../components/Button";
import "./RecipePage.scss";
import { RecipeContext } from "../../components/utils/contexts/RecipeContext";

const RecipePage: FC<{}> = () => {
    const recipeContext = useContext(RecipeContext);
    let { recipeId } = useParams();

    let openRecipe;
 
    if (!recipeContext.recipes || Object.keys(recipeContext.recipes).length === 0) {
        openRecipe = recipeData[0]
    } else {
        openRecipe = Object.values(recipeContext.recipes).filter(item => {
            return item.recipe.shareAs.includes(recipeId as string);
        })[0];
    }

    const kkal = openRecipe.recipe.totalNutrients?.ENERC_KCAL.quantity;
    const fat = openRecipe.recipe.totalNutrients?.FAT.quantity;
    const prot = openRecipe.recipe.totalNutrients?.PROCNT.quantity;
    const carbs = openRecipe.recipe.totalNutrients?.CHOCDF.quantity;
    const tags = openRecipe.recipe.healthLabels?.slice(0,6)

    return (
        <Container>
            <div className='recipe-page'>
                <div className="recipe-page__side">
                    <div className="recipe__tags">
                        {tags?.map((tag, i) => <span className="recipe__tag" key={ i}>{ tag}</span>)}
                    </div>
                    <span className='recipe__title'>{openRecipe.recipe.label}</span>
                    <img src={openRecipe.recipe?.image} alt={openRecipe.recipe.url}
                        className="recipe__img" />
                    <div className="recipe__time">Cook time:
                        {openRecipe.recipe.totalTime ? ` ${openRecipe.recipe.totalTime}` : ' 5'} min</div>
                </div>

                <div className="recipe-page__main">
                    <div className="recipe__nutrition">
                        <span className="recipe__nutrition_kkal">
                            Calories: { kkal ?  `${Math.ceil(kkal)} kkal` :'100 kkal' }
                        </span>
                        <span className="recipe__nutrition_fat">
                            Fat: { fat ?  `${Math.ceil(fat)} g` :'10 g' }
                        </span>
                        <span className="recipe__nutrition_prot">
                            Protein: { prot ?  `${Math.ceil(prot)} g` :'10 g' }
                        </span>
                        <span className="recipe__nutrition_carbs">
                            Carbs: { carbs ?  `${Math.ceil(carbs)} g` :'10 g' }
                        </span>
                    </div>
                    <div className="recipe__ingredients">
                        <div className="recipe__ingredient_title">Ingredients:</div>
                        {openRecipe.recipe.ingredientLines?.map((item, i) => {
                            return <li className="recipe__ingredient" key={ i}>{item}</li>
                        })}
                    </div>
                    <div className="recipe__notes">
                        <div className="recipe__notes_title">NOTES</div>
                        <ul className="recipe__notes_list">
                            <li className="recipe__notes_text">Serve this dip with your favorite fresh vegetable sticks! I recommend cucumber, carrot, bell peppers and cherry tomatoes. Itâ€™s also great with crackers!</li>
                            <li className="recipe__notes_text">The dip will keep in an airtight container in the fridge for up to 3 days. It will taste even more delicious after the flavors have time to meld together!</li>
                        </ul>
                    </div>
                </div>
            </div>
            <div className="recipe-page__button">
                <Link to="/recipes/alcohol-free"><Button content="View All Recipes"
                    type={Button.TYPES.DEFAULT} />
                </Link>
            </div>
        </Container>
    );
};

export default RecipePage;
