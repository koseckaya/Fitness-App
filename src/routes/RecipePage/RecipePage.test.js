import React from 'react';
import renderer from 'react-test-renderer';

import RecipePage from './RecipePage';


it('renders correctly', () => {
    const tree = renderer
        .create(<RecipePage />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});