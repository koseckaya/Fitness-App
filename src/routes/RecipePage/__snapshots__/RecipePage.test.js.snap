// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipePage component renders correctly 1`] = `
<div
  className="container "
>
  <div
    className="recipe-page"
  >
    <div
      className="recipe-page__side"
    >
      <div
        className="recipe__tags"
      >
        <span
          className="recipe__tag"
        >
          Sugar-Conscious
        </span>
        <span
          className="recipe__tag"
        >
          Low Potassium
        </span>
        <span
          className="recipe__tag"
        >
          Kidney-Friendly
        </span>
        <span
          className="recipe__tag"
        >
          Vegetarian
        </span>
        <span
          className="recipe__tag"
        >
          Pescatarian
        </span>
        <span
          className="recipe__tag"
        >
          Dairy-Free
        </span>
      </div>
      <span
        className="recipe__title"
      >
        Naomi Duguid's Fried Noodles
      </span>
      <img
        alt="http://www.seriouseats.com/recipes/2012/10/naomi-duguids-fried-noodles.html"
        className="recipe__img"
        src="https://edamam-product-images.s3.amazonaws.com/web-img/041/04158b5869398c899942336274f0e0f7.jpg?X-Amz-Security-Token=IQoJb3JpZ2luX2VjENf%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLWVhc3QtMSJHMEUCIE8DxlYh37WHsQK19p%2Fm3VfXbYKdruK7FMWBSsc6zVubAiEAwPZ0DTkt3nNSfNS3xCFkf1lilGeCVPGDu3til9alP4kqzAQIXxAAGgwxODcwMTcxNTA5ODYiDAkL7y2aeqWG3ke7SiqpBLqI5dq%2FBPsanAWaj%2FwK4%2FyFbBG6XNGzBKb7tAx8cviM8DwGg285IF1Sg1Vx7Hic37EDSat1brvsCGYl5CQq1TRYZuKbNLVq6O43UNQXOLtiJYziQuhMybWw1F5zUT%2B1nlpi2LYqsKABHoSEfTHS9GfbjAfQhp19ftk0AQqJHh6sQdpUb3zzM2VXgA6HifOMqU8CM%2Bs9LnMC5zlJHRNqN6JFmZsyUyDx6Kc898YMcR06hOwZXQjQwp%2BXJvVogyHIuAZajWelR6V3LxRTyUegpz8UF2wpq4op%2Fr%2FlsKZe37rJuqhxQ8biyLbyzWbVFWq1iSwb%2BYGirpPa%2FTQbNQ%2FWsDAnTSTSbMuAfuX6VCGkqoEzUHRXn353LG933Oqp%2FAlKdsk8%2FlquOUVrAFgDP8drv0xiTeGuCQNljj2iZO0I1TvyZf1gGSAhRJ1ygV9vNbVNZoPXe2ZqzJ6gp12vzr5KXI4avPXyoMKfpHa157x59HjRncL6D0cm7yy3bZrxPAfpDIyQR%2BvLXT6sCA%2Boo75t7PBa%2FV6lck14mrnseov1lzod3OBhA4xoMp%2Bh0EGhMOaA0y049t%2FXeCUEp6jQrw9KGcvP8j%2BUJyfE7nYHXBoWV5NHDVxlXoQ5Vi%2Fg9dkqvMKukduj05Qe4XevUddBu%2FmCzDJTVLJMDVCtljmw6xYenR%2FUCEsVxaHMCuBOa%2FJ75yS%2FPsxSFjdCneONmM%2F73AcXnVnv5rMu2Qw0WOww8LaJnwY6qQGsEzaX%2BQi0Y3BdoW6GnW3%2FtlILc6l8Z4g4VoaHAD0IhbcLFIFh6bPFhFMdMD%2B1XkL7Pn77tqgB801q3OuJobRUWjk18xGUayil8dRgfTEhevi2QTfh41lnVnrnBnobrRBk0V7ZG2EWQ8enh1AXtGdk4iiZD1eTzyilT12jaahgNM4PZaQ2eHTTUeWKEIoqeM%2Fc6zSJ2MNF5M5ZWqnSG4bcPsUXVSxLlHRf&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20230207T150741Z&X-Amz-SignedHeaders=host&X-Amz-Expires=3600&X-Amz-Credential=ASIASXCYXIIFOMC2ZVSB%2F20230207%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Signature=fe8b6ae85466c4bffd9afb033f019a7ff865bdf67870b4bfbc7906e10cc8c621"
      />
      <div
        className="recipe__time"
      >
        Cook time:
         20
         min
      </div>
    </div>
    <div
      className="recipe-page__main"
    >
      <div
        className="recipe__nutrition"
      >
        <span
          className="recipe__nutrition_kkal"
        >
          Calories: 
          197 kkal
        </span>
        <span
          className="recipe__nutrition_fat"
        >
          Fat: 
          8 g
        </span>
        <span
          className="recipe__nutrition_prot"
        >
          Protein: 
          6 g
        </span>
        <span
          className="recipe__nutrition_carbs"
        >
          Carbs: 
          28 g
        </span>
      </div>
      <div
        className="recipe__ingredients"
      >
        <div
          className="recipe__ingredient_title"
        >
          Ingredients:
        </div>
        <li
          className="recipe__ingredient"
        >
          1 cup dried egg noodles
        </li>
        <li
          className="recipe__ingredient"
        >
          Peanut oil, for frying
        </li>
      </div>
      <div
        className="recipe__notes"
      >
        <div
          className="recipe__notes_title"
        >
          NOTES
        </div>
        <ul
          className="recipe__notes_list"
        >
          <li
            className="recipe__notes_text"
          >
            Serve this dip with your favorite fresh vegetable sticks! I recommend cucumber, carrot, bell peppers and cherry tomatoes. Itâ€™s also great with crackers!
          </li>
          <li
            className="recipe__notes_text"
          >
            The dip will keep in an airtight container in the fridge for up to 3 days. It will taste even more delicious after the flavors have time to meld together!
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="recipe-page__button"
  >
    <a
      href="/recipes/alcohol-free"
      onClick={[Function]}
    >
      <div
        className="button  "
      >
        View All Recipes
      </div>
    </a>
  </div>
</div>
`;
